helper/resource/testing_sets.go

```diff
@@ -64,17 +85,47 @@ func TestCheckTypeSetElemNestedAttrs(name, attr string, values map[string]string
 	}
 }
 
-// TestMatchTypeSetElemNestedAttrs is a TestCheckFunc similar to TestCheckTypeSetElemNestedAttrs
-// with the exception that it verifies that an element matches a *regexp.Regexp.
+// TestMatchTypeSetElemNestedAttrs ensures a subset map of values, compared by
+// regular expressions, is stored in state for the given name and key
+// combination of attributes nested under a list or set block. Use this
+// TestCheckFunc in preference over non-set variants to simplify testing code
+// and ensure compatibility with indicies, which can easily change with schema
+// changes. State value checking is only recommended for testing Computed
+// attributes and attribute defaults.
+//
+// For managed resources, the name parameter is combination of the resource
+// type, a period (.), and the name label. The name for the below example
+// configuration would be "myprovider_thing.example".
+//
+//     resource "myprovider_thing" "example" { ... }
+//
+// For data sources, the name parameter is a combination of the keyword "data",
+// a period (.), the data source type, a period (.), and the name label. The
+// name for the below example configuration would be
+// "data.myprovider_thing.example".
+//
+//     data "myprovider_thing" "example" { ... }
+//
+// The key parameter is an attribute path in Terraform CLI 0.11 and earlier
+// "flatmap" syntax. Keys start with the attribute name of a top-level
+// attribute. Use the sentinel value '*' to replace the element indexing into
+// a list or set. The sentinel value can be used for each list or set index, if
+// there are multiple lists or sets in the attribute path.
 //
-// You may check for unset keys, however this will also match keys set to empty
-// string. Please provide a map with at least 1 non-empty value e.g.
+// The values parameter is the map of attribute names to regular expressions
+// for matching attribute values expected to be nested under the list or set.
 //
-//   map[string]*regexp.Regexp{
-//	     "key1": regexp.MustCompile("value"),
-//       "key2": regexp.MustCompile(""),
-//   }
+// You may check for unset nested attributes, however this will also match keys
+// set to empty string. Use a map with at least 1 non-empty value.
```

Comment:
`set to empty string` -> `set to an empty string`

Reply:
```suggestion
// set to an empty string. Use a map with at least 1 non-empty value.
```